### PR TITLE
Feature: separate vault value for db migration

### PR DESCRIPTION
**Pull request by doncem**
In order to migrate from deprecated features, particularly [this](https://github.com/hmcts/cnp-jenkins-library#secrets-for-functional--smoke-testing) guideline on _howto_, some projects might use multiple vaults to load secrets - as suggested in the link previously. This feature requires `vaultName` to be removed from terraform output file. Which then causes null pointer in case project has db migration step in the pipeline.

Introducing separate config just for that and securing from null value

### CODE REVIEW

File: vars/sectionDeployToEnvironment.groovy [hunk 1]
```diff
@@ -52,7 +52,10 @@ def call(params) {
             if (config.migrateDb) {
               stage("DB Migration - ${environment}") {
                 pcr.callAround("dbmigrate:${environment}") {
-                  builder.dbMigrate(tfOutput.vaultName.value, tfOutput.microserviceName.value)
+                  builder.dbMigrate(
+                    tfOutput.vaultName ? tfOutput.vaultName.value : config.dbMigrationVaultName,
+                    tfOutput.microserviceName.value
```
Comment (timja) [inline vars/sectionDeployToEnvironment.groovy]: hmm we need to do something about this as well so we can stop relying on the tfoutput,
shouldn't this just be the component parameter?

File: src/uk/gov/hmcts/contino/AppPipelineDsl.groovy [hunk 2]
```diff
@@ -29,8 +29,9 @@ class AppPipelineDsl extends CommonPipelineDsl implements Serializable {
     WarningCollector.addPipelineWarning("deprecated_set_vault_name", "setVaultName() is deprecated, see https://github.com/hmcts/cnp-jenkins-library#secrets-for-functional--smoke-testing ", new Date().parse("dd.MM.yyyy", "27.08.2019"))
   }
 
-  void enableDbMigration() {
+  void enableDbMigration(String dbMigrationVaultName = "") {
```
Comment (adusumillipraveen) [inline src/uk/gov/hmcts/contino/AppPipelineDsl.groovy]: Should we make it a non-breaking change, just add it to pipelinewarnings ? something like : https://github.com/hmcts/cnp-jenkins-library/blob/a50fd5216c1b42064a9557e59b479509bff91d27/src/uk/gov/hmcts/contino/AppPipelineDsl.groovy#L29

Comment (timja) [reply src/uk/gov/hmcts/contino/AppPipelineDsl.groovy]: isn't this non breaking already? a default has been added

Comment (adusumillipraveen) [reply src/uk/gov/hmcts/contino/AppPipelineDsl.groovy]: Sorry, i meant "auto-bombing" and "deprecation warning" or do we plan to support both ?

Comment (timja) [reply src/uk/gov/hmcts/contino/AppPipelineDsl.groovy]: yeah makes sense to add it 👍

**Comment by doncem** [comment]
Feedback: DB migration step has passed, [link here](https://sandbox-build.platform.hmcts.net/job/HMCTS_Sandbox_Platform/job/send-letter-service/job/tmp%252Ffix/3)

File: src/uk/gov/hmcts/contino/AppPipelineDsl.groovy (hunk 2 above)
Comment (timja) [reply src/uk/gov/hmcts/contino/AppPipelineDsl.groovy]:
```suggestion
  void enableDbMigration(String dbMigrationVaultName = "") {
    WarningCollector.addPipelineWarning("deprecated_enable_db_migration_no_vault)", "enableDbMigration() is deprecated, please use enableDbMigration("<vault-name>")", new Date().parse("dd.MM.yyyy", "05.09.2019"))  
```

File: src/uk/gov/hmcts/contino/AppPipelineDsl.groovy [hunk 3]
```diff
@@ -29,8 +29,10 @@ class AppPipelineDsl extends CommonPipelineDsl implements Serializable {
...
+    WarningCollector.addPipelineWarning("deprecated_enable_db_migration_no_vault)", "enableDbMigration() is deprecated, please use enableDbMigration(<vault-name>)", new Date().parse("dd.MM.yyyy", "05.09.2019"))
```
Comment (timja) [inline src/uk/gov/hmcts/contino/AppPipelineDsl.groovy]: this needs an if statement around it
 ```suggestion
    if (dbMigrationVaultName == "") {
      WarningCollector.addPipelineWarning("deprecated_enable_db_migration_no_vault)", "enableDbMigration() is deprecated, please use enableDbMigration(<vault-name>)", new Date().parse("dd.MM.yyyy", "05.09.2019"))
```

Comment (doncem) [reply src/uk/gov/hmcts/contino/AppPipelineDsl.groovy]: aye, thanks

**Pull request merged**